homeassistant/components/comelit/humidifier.py

```diff
@@ -0,0 +1,215 @@
+        self._attr_unique_id = f"{config_entry_entry_id}-{device.index}-{device_class}"
+        self._attr_name = f"{device_class.capitalize()}"
+        self._attr_device_info = coordinator.platform_device_info(device, device_class)
+        self._attr_device_class = (
+            HumidifierDeviceClass.DEHUMIDIFIER
+            if device_class == HumidifierDeviceClass.DEHUMIDIFIER.value
+            else HumidifierDeviceClass.HUMIDIFIER
+        )
+
+    @property
+    def _humidifier(self) -> list[Any]:
+        """Return humidifier device data."""
+        # CLIMATE has a 2 item tuple:
+        # - first  for Clima
+        # - second for Humidifier
+        return self.coordinator.data[CLIMATE][self._device.index].val[1]
+
+    @property
+    def _api_mode(self) -> str:
+        """Return device mode."""
+        # Values from API: "O", "L", "U"
+        return self._humidifier[2]
+
+    @property
+    def _api_active(self) -> bool:
+        "Return device active/idle."
+        return self._humidifier[1]
+
+    @property
+    def _api_automatic(self) -> bool:
+        """Return device in automatic/manual mode."""
+        return self._humidifier[3] == HumidifierComelitMode.AUTO
+
+    @property
+    def _is_dehumidifier(self) -> bool:
+        """Return true if device is set as dehumidifier."""
+        return (
+            self._api_mode == HumidifierComelitMode.LOWER
+            and self._attr_device_class == HumidifierDeviceClass.DEHUMIDIFIER
+        )
+
+    @property
+    def _is_humidifier(self) -> bool:
+        """Return true if device is set as humidifier."""
+        return (
+            self._api_mode == HumidifierComelitMode.UPPER
+            and self._attr_device_class == HumidifierDeviceClass.HUMIDIFIER
+        )
+
+    @property
+    def target_humidity(self) -> int:
+        """Set target humidity."""
+        return self._humidifier[4] / 10
+
+    @property
+    def current_humidity(self) -> int:
+        """Return current humidity."""
+        return self._humidifier[0] / 10
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true is humidifier is on."""
+        return self._is_dehumidifier or self._is_humidifier
+
+    @property
+    def mode(self) -> str | None:
+        """Return current mode."""
+        return MODE_AUTO if self._api_automatic else MODE_NORMAL
+
+    @property
+    def action(self) -> HumidifierAction | None:
+        """Return current action."""
+
+        if self._api_mode == HumidifierComelitMode.OFF:
+            return HumidifierAction.OFF
+
+        if self._api_active and self._is_dehumidifier:
+            return HumidifierAction.DRYING
+
+        if self._api_active and self._is_humidifier:
+            return HumidifierAction.HUMIDIFYING
+
+        return HumidifierAction.IDLE
+
+    async def async_set_humidity(self, humidity: int) -> None:
+        """Set new target humidity."""
+        if self.mode == HumidifierComelitMode.OFF:
+            return
```

Comment:
This should raise a HomeAssistantError (or service error, i don't remember). You are ignoring a request.

Reply:
I did the same in climate ;-)
Let me check how to handle properly then

Reply:
Fixed